react/Button/Button.js

```diff
@@ -10,9 +10,13 @@ export default class Button extends Component {
   static displayName = 'Button';
 
   static propTypes = {
-    color: PropTypes.oneOf([
-      'pink', 'blue', 'gray', 'transparent'
-    ]).isRequired,
+    /* eslint-disable consistent-return */
+    color: (props, propName, componentName) => {
+      if (!props.group && ['pink', 'blue', 'gray', 'transparent'].indexOf(props.color) === -1) {
+        return new Error(`When ${componentName} doesn't not have \`group\`, it should have an \`color\`.`);
```

Comment:
The fact that we need to write code like this is making me reconsider this pattern 🤔